File: src/main/java/jp/osak/haggledehaghag/viewmodel/GameView.java
package jp.osak.haggledehaghag.viewmodel;

import jp.osak.haggledehaghag.model.Game;

public record GameView(
        int id,
        String title
) {
    public static GameView of(final Game game) {
        return new GameView(game.id(), game.title());
    }
}

File: src/main/java/jp/osak/haggledehaghag/viewmodel/GameView.kt
package jp.osak.haggledehaghag.viewmodel

import jp.osak.haggledehaghag.model.Game

data class GameView(
    val id: Int,
    val title: String,
    val gameKey: String,
) {
    constructor(game: Game) : this(game.id, game.title, game.gameKey)
}
